python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -184,25 +185,64 @@ def _get_p2p_data(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_
         msg = f"{self.data_model}.p2p_links must have either 'interfaces' or 'port_channel' with correct members set."
         raise AristaAvdInvalidInputsError(msg)
 
-    def _get_common_interface_cfg(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_link: T_P2pLinksItem, p2p_link_data: dict) -> dict:
+    def _get_ptp_config_interface(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_link: T_P2pLinksItem, output_type: type[T_Ptp]) -> T_Ptp:
         """
-        Return partial structured_config for one p2p_link.
+        Return ptp config for one p2p_link.
+
+        Covers common config that is applicable to both port-channels and ethernet interfaces.
+        This config will only be used on the main interface - so not port-channel members.
+        """
+        ptp_config = output_type()
+        if p2p_link.ptp.enabled:
```

Comment:
let's revert the logic
```suggestion
if not p2p_link.ptp.enabled:
  return ptp_config
```

Reply:
fixed